spring-integration-core/src/main/java/org/springframework/integration/aggregator/CorrelatingMessageBarrier.java

```diff
@@ -103,12 +110,14 @@ private Object getLock(Object correlationKey) {
 
 	@SuppressWarnings("unchecked")
 	@Override
+	@Nullable
 	public Message<Object> receive() {
 		for (Object key : this.correlationLocks.keySet()) {
 			Object lock = getLock(key);
 			synchronized (lock) {
 				MessageGroup group = this.store.getMessageGroup(key);
 				//group might be removed by another thread
+				Assert.notNull(this.releaseStrategy, "'releaseStrategy' must not be null");
```

Comment:
Not the task right now, but I suggestion to optimize this a bit and have this assert and the one in the `handleMessageInternal()` for  the `correlationStrategy` in the `onInit()` impl.
And have both pros as `	@SuppressWarnings("NullAway.Init")`.
Something to do on a spare time 😄